src/mapping/query/functions.rs

```diff
@@ -49,24 +140,54 @@ impl Function for ToStringFn {
                 _ => Value::Bytes(v.as_bytes()),
             }),
             Err(err) => match &self.default {
-                Some(v) => Ok(v.clone()),
+                Some(v) => v.execute(ctx),
                 None => Err(err),
             },
         }
     }
+
+    fn parameters() -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kinds: &[],
+                required: true,
+            },
+            Parameter {
+                keyword: "default",
+                kinds: &[],
+                required: false,
+            },
+        ]
+    }
+}
+
+impl TryFrom<ArgumentList> for ToStringFn {
+    type Error = String;
+
+    fn try_from(mut arguments: ArgumentList) -> Result<Self> {
+        let query = arguments.required("value")?;
```

Comment:
This _should_ never return an error, since the parser validates at compile-time that required arguments are provided. I opted to still make this explicit though, by having it return a `Result<Argument>` as opposed to `optional`, which returns `Option<Argument>`.